services/transaction/shared/transaction.js

```diff
@@ -82,7 +111,44 @@ const createMultisignatureTx = async inputTransaction => {
 	return transaction;
 };
 
+const updateMultisignatureTx = async transactionPatch => {
+	const multisignatureTxDB = await getMultiSignatureTxIndex();
+	const multisigSignaturePool = await getMultisigSignaturePool();
+	const transaction = {
+		data: [],
+		meta: {},
+	};
+	const { serviceId, signatures } = transactionPatch;
+
+	// Find the transaction from the pool
+	const [pooledTransaction] = await multisignatureTxDB.find({ serviceId });
+	if (!pooledTransaction) throw new NotFoundException(`Transaction with serviceId: ${serviceId} does not exist in the pool`);
+
+	try {
+		// Validate and add the signatures into the pool
+		// TODO: Add transactional support and validations
+		await BluebirdPromise.map(
+			signatures,
+			async signature => multisigSignaturePool.upsert({
+				serviceId: transactionPatch.serviceId,
+				signature: JSON.stringify(signature),
+			}),
+			{ concurrency: signatures.length },
```

Comment:
Auto-commit seems to be fine in that case. It might be beneficial to add `concurrency: 1` to prevent deadlocks, as there is no fallback job for that scenario.

Reply:
Updated!

However, `{ concurrency: signatures.length }` should never cause deadlocks since with the `PATCH` request, we do not update any existing rows rather only adds new entries to the signature pool.